libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java

```diff
@@ -1491,4 +1506,89 @@ public static String markMediaFailed(Context context, @NonNull String postConten
         }
         return postContent;
     }
+
+    public static boolean hasMediaItemsMarkedUploading(Context context, @NonNull String postContent) {
+        return hasMediaItemsMarkedWithTag(context, postContent, ATTR_STATUS_UPLOADING);
+    }
+
+    public static boolean hasMediaItemsMarkedFailed(Context context, @NonNull String postContent) {
+        return hasMediaItemsMarkedWithTag(context, postContent, ATTR_STATUS_FAILED);
+    }
+
+    private static boolean hasMediaItemsMarkedWithTag(Context context, @NonNull String postContent, String tag) {
+        // fill in Aztec with the post's content
+        AztecText content = new AztecText(context);
+        content.fromHtml(postContent);
+
+        // get all items with "uploading" class
+        AztecText.AttributePredicate uploadingPredicate = getPredicateWithClass(tag);
+
+        List<AztecAttributes> attrs = content.getAllElementAttributes(uploadingPredicate);
```

Comment:
some optimization could be done there by adding a new method in AztecText, but yeah I don't think it's necessary